### ISSUE TITLE
Support for sd card volme create

### ISSUE THREAD

**Issue by Thewisem** [hardcore-sushi/DroidFS#34]
Could you add support for sd card vlume write?

**Comment by hardcore-sushi** [comment]
AFAIK, a normal app can only write to SD card through the Android [ContentProvider](https://developer.android.com/guide/topics/providers/content-providers) API, which is not compatible with DroidFS. Android 11 introduced the [MANAGE_EXTERNAL_STORAGE](https://developer.android.com/reference/android/Manifest.permission#MANAGE_EXTERNAL_STORAGE) permission which seems to solve the problem. However it's not available for versions below Android 11.

**Comment by Thewisem** [comment]
Couldn't you use the storage access framework to act as a location to external storage to keep your encrypted volume.

**Comment by hardcore-sushi** [comment]
No, unless you find a way to use SAF with paths, without API calls.

**Comment by Thewisem** [comment]
I was wondering you could do it like keepass DX. They used SAF and allowed users to create databases on SD card. Also read and write was possible. Maybe ask the devs there.

**Comment by Thewisem** [comment]
I mean this is not like partitioning your drive as it is creating encrypted files for the new volume you created. So I thought since both use the same methodology, then it was possible.

**Comment by hardcore-sushi** [comment]
KeepassDX and other apps use SAF from the JVM, via API calls. DroidFS needs to access file system from native code but native code can only use SAF through file descriptors. Gocryptfs uses paths. It's not compatible.

See [#6](https://github.com/hardcore-sushi/DroidFS/issues/6#issuecomment-721052808) for more details.

**Comment by retardTonic** [comment]
Any app with `WRITE_EXTERNAL_STORAGE` permission can write to  `/{path to SD card}/Android/data/{app package name}/`. Not sure about OTG, though.

Workaround (tested on Android 10):

- Create directory `/storage/1234-5678/Android/data/sushi.hardcore.droidfs/files`, replace numbers with UUID.
- Specify this directory or any subdirectory as volume path.
- File picker button inside volume creation and password change activities would not work, so you'll have to type in path manually or through clipboard.

An additional check in function [isTreeUriOnPrimaryStorage](https://github.com/hardcore-sushi/DroidFS/blob/a4db2740a190862ad128ba8083fc77fe35c9e54b/app/src/main/java/sushi/hardcore/droidfs/util/PathUtils.kt#L97) should be implemented, I guess?

**Comment by hardcore-sushi** [comment]
You're right, but DroidFS can't access arbitrary files on SD card like `/{path to SD card}/MyEncryptedVolume`. This is why I disabled SD card usage. Maybe I could still allow `Android/data/sushi.hardcore.droidfs/files`.
